File: C++/minimum-number-of-vertices-to-reach-all-nodes.cpp
class Solution {
public:   
    
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        vector<int> ans;
        
        set<int> from;
        set<int> to;
        
        for (auto e: edges) {
            from.insert(e[0]);
            to.insert(e[1]);
        }
        
        for (int f: from) {
            if (!to.count(f)) {
                ans.push_back(f);
            }
        }
        
        return ans;
    }
    
};

File: JavaScript/minimum-number-of-vertices-to-reach-all-nodes.js
/**
 * @param {number} n
 * @param {number[][]} edges
 * @return {number[]}
 */
var findSmallestSetOfVertices = function(n, edges) {
    const incomingEdge = Array(n).fill(false)
    for (const [_, to] of edges) {
        incomingEdge[to] = true
    }
    const ans = []
    for (let i = 0; i < n; i++) {
        if (!incomingEdge[i]) {
            ans.push(i)
        }
    }
    return ans
};